tidy3d/components/types_extra.py

```diff
@@ -0,0 +1,11 @@
+from typing import Any
```

Comment:
also, a few of the other new files are missing them as well.

Reply:
So it's actually intentional. I will comment about this since it relates to Momchil's comment in https://github.com/flexcompute/tidy3d/pull/1376

Reply:
thanks, still would be good to put a module level docstring though (I thought ruff would have caught these)

Reply:
So I think if we add it to the top level module, then it's no longer on-the-fly importing, since these types are only used when a function imports them - that's why I did not add it in. 

The idea was that we have sections of code that are not part of the core but can be called within the core functions - that's how I we got the import speedups. What do you think?